### PR TITLE
Adding Capacity Provider Support, part 1: Fargate/Fargate Spot

### PR DESCRIPTION
**Pull request by adamjkeller**
In this update, we will implement capacity providers in two ways, via two pull requests:

(this pull request)
1) We will implement a split deployment strategy for Fargate tasks to deploy half as Fargate on demand, and the other half as Fargate Spot.

(next pull request)
2) We will implement an EC2 backed strategy, deploy a service that will scale out past the current capacity, and watch as cluster autoscaling scales the backend EC2 nodes to meet the demand.

### CODE REVIEW

File: content/capacity_providers/tabs/fargate.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+```bash
+aws ecs put-cluster-capacity-providers \
+--cluster container-demo \
+--capacity-providers FARGATE FARGATE_SPOT \
+--default-capacity-provider-strategy \
+capacityProvider=FARGATE,weight=1,base=1 \
+capacityProvider=FARGATE_SPOT,weight=4
+```
+
+With this command, we're adding the Fargate and Fargate Spot capacity providers to our ECS Cluster. Let's break it down by each input:
```
[the remark this answers is not in the document]
Comment (adamMeeruTech) [reply content/capacity_providers/tabs/fargate.md:22]: lol, good catch. Changing

File: content/capacity_providers/capacityprovider_primer.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+When you run a task or create a service, you may either use the cluster's default capacity provider strategy or specify a capacity provider strategy that overrides the cluster's default strategy. 
+
+For more information, see the documentation [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-capacity-providers.html)
+
+
+## Concepts
+
+Capacity providers can be used for ECS Fargate tasks, and ECS EC2 backed tasks.
+
+- Fargate capacity providers enable you to use both Fargate and Fargate Spot capacity with your Amazon ECS tasks. For more information, see [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-capacity-providers.html)
```
[the remark this answers is not in the document]
Comment (adamjkeller) [reply content/capacity_providers/capacityprovider_primer.md]: Much clearer, made the change.

File: content/capacity_providers/tabs/fargate.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamjkeller) [reply content/capacity_providers/tabs/fargate.md]: sometimes I forget how to use words. Thank you :-D

File: content/capacity_providers/capacityprovider_primer.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+- Amazon ECS on EC2 capacity providers enable customers to use Cluster Auto Scaling, allowing the focus of the customer to shift from managing autocaling the backend infrastructure, to focusing on supporting the application.
+for more information on Amazon ECS Cluster Auto Scaling, see [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-auto-scaling.html)
```
[the remark this answers is not in the document]
Comment (adamjkeller) [reply content/capacity_providers/capacityprovider_primer.md]: Much clearer, made the change.

File: content/capacity_providers/tabs/fargate.md [hunk 4]
```diff
@@ -145,8 +145,8 @@ Here's what we accomplished in this section of the workshop:
...
+- While this was just an example, this could translate to many real world use cases. By simply setting the base and weights between Fargate and Fargate Spot, we can take advantage of the cost savings of Fargate Spot in our every day workloads. Of course, it's important to understand that Spot tasks can be terminated at any time (for more information, [check out the official AWS documentation.](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-capacity-providers.html#fargate-capacity-providers-termination)), when capacity requirements change. That is why we set the default strategy to be a mix of Fargate and Fargate spot to ensure that if spot tasks are terminated, we still have our minimum, desired amount of tasks running on Fargate.
+- The strategy we chose was to use a mix of strategies (Fargate and Fargate Spot). You can also stick to one strategy (Fargate or Fargate Spot), and this would be defined when you deploy your service or as the default for the cluster.
```
Comment (brentley) [inline content/capacity_providers/tabs/fargate.md:149]: one last thing... strategy of strategies kind of bothers me... how about "the setting we chose was to use a mix of strategies"

Comment (adamjkeller) [reply content/capacity_providers/tabs/fargate.md]: excellent, good catch. Done!

**Pull request merged**